sdks/common/runner-api/src/main/proto/beam_runner_api.proto

```diff
@@ -0,0 +1,450 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Protocol Buffers describing the Runner API, which is the runner-independent,
+ * SDK-independent definition of the Beam model.
+ */
+
+syntax = "proto3";
+
+package org.apache.beam.runner_api.v1;
+
+option java_package = "org.apache.beam.sdks.common.runner_api.v1";
+option java_outer_classname = "RunnerApi";
+
+import "google/protobuf/any.proto";
+import "google/protobuf/timestamp.proto";
+
+// A Pipeline contains the entire hierarchical graph of transforms as well
+// as descriptors for all PCollections.
+//
+// The hierarchical graph itself is an independent layer from the payloads
+// specific to the Runner API, for ease of reuse by e.g. the Fn API.
+message Pipeline {
+
+  // (Required) A map from pipeline-scoped id to graph node.
+  //
+  // Each node is required to contain a PTransform specification.
+  map<int64, GraphNode> transform_nodes = 1;
+
+  // (Required) A map from pipeline-scoped id to PCollection.
+  map<int64, PCollection> pcollections = 2;
+
+  // (Required) A map from pipeline-scoped id to WindowingStrategy.
+  map<int64, WindowingStrategy> windowing_strategies = 3;
+
+  // (Required) A map from pipeline-scoped id to Coder.
+  map<int64, Coder> coders = 4;
+
+  // (Required) A map from pipeline-scoped id to Environment.
+  map<int64, Environment> environments = 5;
+
+  // (Optional) Static display data for the pipeline.
```

Comment:
These could be made (Required) such that consuming runners don't need to check it's existence before reading it. SDK's which don't yet collect display data can pass the empty collections.

Same comment applies to other `display_data` fields.